### DISCUSSION TITLE
Where can I select a physical CD drive?

### DISCUSSION THREAD

**Discussion by ghost** [86Box/86Box#1229] [Q&A] [answered]
![image](https://user-images.githubusercontent.com/31448055/80335256-c1815b80-8821-11ea-9428-cf7dae0e091b.png)
Hello. I apologize if this is a dumb question, but I need to ask if there is an option for mounting a physical CD drive or one mounted using Daemon Tools. Being able to mount ISOs is nice and all, but not being able to use real drives is a bit of a deal breaker for me.

**Comment by nerd73** [comment, marked as answer]
There is no option to mount a virtual or physical drive. Could you please explain why it is a deal breaker for you? As far as I can remember IOCTL (the back-end for that option) was very slow.

**Comment by ghost** [comment]
Ah. Well, thanks for letting me know. I primarily use Daemon Tools to mount MDF/MDS and BIN/CUE images for the games that I rip, and I know that MDF support has been brought up before, but using a virtual drive would eliminate any barriers set by the program in terms of image types.

**Comment by bachimusprime** [comment]
BIN/CUE already works AFAIK.

**Comment by usergithub64** [comment]
Why is the "Mute" function needed in the CD-ROM tab in the top screenshot? What does this function do?

**Comment by ghost** [comment]
CD drives of that era and some late ones had Audio CD functionality. They had a cable were Audio CD data passed and were processed straight to the sound card. Some had volume control too allowing you to control well, volume.

**Comment by ghost** [comment]
> 
> 
> Ah. Well, thanks for letting me know. I primarily use Daemon Tools to mount MDF/MDS and BIN/CUE images for the games that I rip, and I know that MDF support has been brought up before, but using a virtual drive would eliminate any barriers set by the program in terms of image types.

Mount the *.cue image. It works.

**Comment by OBattler** [comment]
Yes, CUE/BIN images work. And I would love to add MDS/MDF support, but I've yet to find some human-readable specification for it. As for passing physical drives, I might look at it, though the best bet would be do them without all the fancy data type stuff, so I'm not sure how good that would be. It would also added more code needed to port when we get around to the Linux port some day.

**Comment by creopard** [comment]
Unfortunately, the specification for Daemon Tools file format MDS/MDF and MDX [does not seem to be an open standard](https://forum.daemon-tools.cc/showthread.php?t=27748). 

referenced for future use: #430

**Reply by claunia** [reply]
Not only not an open standard. They took Alcohol 120% format, and obfuscated it. In reality it is not even a different format, just the same (that is documented), but "encrypted".

**Comment by ppgrainbow** [comment]
Unfortunately, it's not possible to select a physical or virtual CD-ROM/DVD-ROM drive or a floppy diskette drive to use under 86Box at this time. :)

**Reply by Scythifuge** [reply]
Greetings! I was following the project for a bit and was chatting on the discord channel, though I took a break.  I had asked similar questions about adding physical drives and had shared an MS-DOS compatible start-up SATA driver with the hope that it could be used to add a physical SATA optical drive.  Will it ever be possible to select physical drives?  I still have the goal of building a physical computer that uses 86box and DOSBox to emulate various physical, retro machines that can use physical media.

**Comment by angrylion7** [comment]
Well that's a bummer that PCem can capture physical CD-ROM drives but not 86box.

**Reply by OBattler** [reply]
We're going to look into it after the Linux port is done and we can then test such functionality on all platforms.

**Comment by Scythifuge** [comment]
Hello! What is the current status of allowing physical optical drives? I am also curious about other physical drives such as mounting memory cards or other USB drives.
